db/rdbms/src/main/resources/mapper/JobMapper.xml

```diff
@@ -12,18 +12,200 @@
 
 <mapper namespace="io.camunda.db.rdbms.sql.JobMapper">
 
+  <select id="count" resultType="java.lang.Long">
+    SELECT COUNT(*)
+    FROM ${prefix}JOB
+    <where>
+      <include refid="io.camunda.db.rdbms.sql.JobMapper.searchFilter" />
+    </where>
+  </select>
+
+  <!-- default search statement for databases supporting LIMIT/OFFSET-->
+  <select id="search"
+    parameterType="io.camunda.db.rdbms.read.domain.JobDbQuery"
+    resultMap="io.camunda.db.rdbms.sql.JobMapper.searchResultMap"
+    statementType="PREPARED">
+    SELECT * FROM (
+      SELECT
+        JOB_KEY,
+        PARTITION_ID,
+        PROCESS_INSTANCE_KEY,
+        ELEMENT_INSTANCE_KEY,
+        PROCESS_DEFINITION_ID,
+        PROCESS_DEFINITION_KEY,
+        TENANT_ID,
+        TYPE,
+        WORKER,
+        STATE,
+        RETRIES,
+        ERROR_MESSAGE,
+        ERROR_CODE,
+        END_TIME,
+        CUSTOM_HEADERS,
+        KIND,
+        ELEMENT_ID,
+        IS_DENIED,
+        DENIED_REASON,
+        LISTENER_EVENT_TYPE,
+        DEADLINE,
+        HAS_FAILED_WITH_RETRIES_LEFT,
+        HISTORY_CLEANUP_DATE
+      FROM ${prefix}JOB
+    <where>
+      <include refid="io.camunda.db.rdbms.sql.JobMapper.searchFilter" />
+    </where>
+
+    ) t
+    <include refid="io.camunda.db.rdbms.sql.Commons.keySetPageFilter" />
+    <include refid="io.camunda.db.rdbms.sql.Commons.orderBy" />
+    <include refid="io.camunda.db.rdbms.sql.Commons.paging" />
+  </select>
+
+  <sql id="searchFilter">
+    <trim prefixOverrides="AND">
```

Comment:
💭 I believe this is only needed when OR queries are supported. Looking at other entities, such as [user tasks](https://github.com/camunda/camunda/blob/0d3a6dfc6ed39b38ac9d3a57452c578cd4c88f85/db/rdbms/src/main/resources/mapper/UserTaskMapper.xml#L144), this is not required in general.